SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/EntityComponentTestUtils.h

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "SpatialView/EntityComponentTypes.h"
+
+namespace SpatialGDK
+{
+
+static const Schema_FieldId EventId = 1;
+static const Schema_FieldId EventIntFieldId = 2;
+
+static const Schema_FieldId TestDoubleFieldId = 1;
+
+inline ComponentData CreateTestComponentData(Worker_ComponentId Id, double Value)
+{
+	ComponentData Data{ Id };
+	Schema_Object* Fields = Data.GetFields();
+	Schema_AddDouble(Fields, TestDoubleFieldId, Value);
+	return Data;
+}
+
+inline ComponentUpdate CreateTestComponentUpdate(Worker_ComponentId Id, double Value)
+{
+	ComponentUpdate Update{ Id };
+	Schema_Object* Fields = Update.GetFields();
+	Schema_AddDouble(Fields, TestDoubleFieldId, Value);
+	return Update;
+}
+
+inline void AddTestEvent(ComponentUpdate* Update, int Value) {
+	Schema_Object* events = Update->GetEvents();
+	Schema_Object* eventData = Schema_AddObject(events, EventId);
+	Schema_AddInt32(eventData, EventIntFieldId, Value);
+}
+
+inline ComponentUpdate CreateTestComponentEvent(Worker_ComponentId Id, int Value) {
+	ComponentUpdate Update{ Id };
+	AddTestEvent(&Update, Value);
+	return Update;
+}
+
+/** Returns true if Lhs and Rhs have the same serialized form. */
+inline bool CompareSchemaObjects(const Schema_Object* Lhs, const Schema_Object* Rhs)
+{
+	const auto Length = Schema_GetWriteBufferLength(Lhs);
+	if (Schema_GetWriteBufferLength(Rhs) != Length)
+	{
+		return false;
+	}
+	const TUniquePtr<unsigned char[]> LhsBuffer = MakeUnique<unsigned char[]>(Length);
```

Comment:
Why the change to unsigned char? I think uint8_t was more explicit and is the type that the worker sdk uses too.

Reply:
Changed to uint8_t